docs/guides/go/realtime-messaging.mdx

```diff
@@ -0,0 +1,298 @@
+  if err != nil {
+    return
+  }
+
+  senderId := ctx.Request.ConnectionID()
+
+  for {
+    connectionId, err := connectionStream.Recv()
+    if err != nil {
+      break
+    }
+
+    if connectionId == senderId {
+      continue
+    }
+
+    message := fmt.Sprintf("%s: %s", senderId, ctx.Request.Message())
+    err = ws.Send(context.TODO(), connectionId, []byte(message))
+    if err != nil {
+      return
+    }
+  }
+})
+```
+
+### Bringing it all together
+
+<details>
+<summary>Your code should look like this:</summary>
+
+```go
+package main
+
+import (
+  "context"
+  "fmt"
+
+  "github.com/nitrictech/go-sdk/nitric"
+  "github.com/nitrictech/go-sdk/nitric/keyvalue"
+  "github.com/nitrictech/go-sdk/nitric/websockets"
+)
+
+func main() {
+  // Create a WebSocket endpoint named "public".
+  ws := nitric.NewWebsocket("public")
+
+  // Initialize a KV store named "connections" with Get, Set, and Delete permissions.
+  connections := nitric.NewKv("connections").Allow(keyvalue.KvStoreGet, keyvalue.KvStoreSet, keyvalue.KvStoreDelete)
+
+  // Handle new WebSocket connections by storing the connection ID in the KV store.
+  ws.On(websockets.EventType_Connect, func(ctx *websockets.Ctx) {
+    err := connections.Set(context.TODO(), ctx.Request.ConnectionID(), map[string]interface{}{
+      "connectionId": ctx.Request.ConnectionID(),
+    })
+    if err != nil {
+      return
+    }
+  })
+
+  ws.On(websockets.EventType_Disconnect, func(ctx *websockets.Ctx) {
+    err := connections.Delete(context.TODO(), ctx.Request.ConnectionID())
+    if err != nil {
+      return
+    }
+  })
+
+  ws.On(websockets.EventType_Message, func(ctx *websockets.Ctx) {
+    connectionStream, err := connections.Keys(context.TODO())
+    if err != nil {
+      return
+    }
+
+    senderId := ctx.Request.ConnectionID()
+
+    for {
+      connectionId, err := connectionStream.Recv()
+      if err != nil {
+        break
+      }
+
+      if connectionId == senderId {
+        continue
+      }
+
+      message := fmt.Sprintf("%s: %s", senderId, ctx.Request.Message())
+      err = ws.Send(context.TODO(), connectionId, []byte(message))
+      if err != nil {
+        return
+      }
+    }
+  })
+
+  nitric.Run()
+}
+```
+
+</details>
+
+Do a quick `go mod tidy` to make sure all new dependencies are resolved.
+
+## Ok, let's run this thing!
+
+Now that you have your WebSocket application defined with handlers for each event, it's time to test it locally. Update your `nitric.yaml` to point to the service at `websockets/main.go`:
+
+```yaml
+services:
+  - match: websockets/*
+    runtime: go
+    start: go run ./$SERVICE_PATH
+```
+
+```bash
+nitric start
+```
+
+Once it starts, the application will be ready to accept WebSocket connections. You can use a WebSocket client like Postman or any other WebSocket tool to test the application. Alternatively, you can use the [Nitric Dashboard](/get-started/foundations/projects/local-development#local-dashboard).
```

Comment:
I think next steps showing how to test it with the Dashboard or a WS client would be handy, with details on how to find the local WS endpoint/URL